File: col_trends-back/src/main/java/com/api/col_trendsback/utils/QueryParameters.java
package com.api.col_trendsback.utils;

public class QueryParameters {
    private String userName;
    private String comment;
    private long idUserQuery;
    private String[] fields;
    private String filter, order;
    private boolean desc;
    public QueryParameters(String[] fields, String filter, String order, boolean desc,String userName) {
        this.fields = fields;
        this.filter = filter;
        this.order = order;
        this.desc = desc;
        this.userName = userName;
    }
    
    public QueryParameters(String userName, long idUserQuery) {
        this.userName = userName;
        this.idUserQuery = idUserQuery; 
    }
    

    public QueryParameters() {
    }

    public String[] getFields() {
        return fields;
    }
    public String getFilter() {
        return filter;
    }
    public String getOrder() {
        return order;
    }
    public boolean isDesc() {
        return desc;
    }

    public long getIdUserQuery() {
        return this.idUserQuery;
    }

    public String getUserName() {
        return userName;
    }

    public String getComment() {
        return comment;
    }
    
}

File: col_trends-front/src/stores/queryParameters.ts
import { json } from 'd3';
import { defineStore } from 'pinia';
import { ref } from 'vue';

interface  QueryParameters {
  comment?: string | null; 
  userName?: string | null ;
  idUserQuery?: number | null;
  fields?: string[];
  filter?: string  | null;
  order?: string | null;
  desc?: boolean;
}

export const useQueryStore = defineStore(
  'queryStore', () => {
    const query = ref<QueryParameters>({fields:[],desc:false});
    const setQuery = (_query: QueryParameters) => {
      query.value = _query;
    }
    const setQueryValue = (_query: QueryParameters) => {
      
      const temp:any = JSON.parse(JSON.stringify({...query.value ,..._query}));
      query.value = temp;
console.log(query.value)
    }
    const resetQueryStore = () => {
      query.value = {fields:[],desc:false};
    }
    return {query, setQuery, setQueryValue,resetQueryStore};
  }
);

